server/crashmanager/serializers.py

```diff
@@ -91,14 +92,12 @@ def create(self, attrs):
             if testcase_ext is None:
                 raise RuntimeError("Must provide testcase extension when providing testcase")
 
-            if testcase_isbinary:
-                testcase = base64.b64decode(testcase)
-
             h = hashlib.new('sha1')
             if testcase_isbinary:
-                h.update(str(testcase))
+                testcase = base64.b64decode(testcase)
+                h.update(testcase)
             else:
-                h.update(repr(testcase))
+                h.update(repr(testcase).encode("utf-8"))
```

Comment:
Can you please check that this doesn't break anything? We had a history with this line breaking due to encoding problems of plaintext tests.